packages/dx-react-scheduler-demos/src/demo-sources/scheduler-editing/material-ui/editing-features.jsx

```diff
@@ -0,0 +1,187 @@
+          )}
+          classes={{ label: classes.formControlLabel }}
+          label={text}
+          key={id}
+          disabled={(id === 'allowDragging' || id === 'allowResizing') && !editingOptions.allowUpdating}
+        />
+      ))}
+    </FormGroup>
+  </div>
+));
+
+export default () => {
+  const [data, setData] = React.useState(appointments);
+  const [editingOptions, setEditingOptions] = React.useState({
+    allowAdding: true,
+    allowDeleting: true,
+    allowUpdating: true,
+    allowDragging: true,
+    allowResizing: true,
+  });
+  const [addedAppointment, setAddedAppointment] = React.useState({});
+  const [appointmentChanges, setAppointmentChanges] = React.useState({});
+  const [editingAppointmentId, setEditingAppointmentId] = React.useState(undefined);
+  const [isAppointmentBeingCreated, setIsAppointmentBeingCreated] = React.useState(false);
+
+  const {
+    allowAdding, allowDeleting, allowUpdating, allowResizing, allowDragging,
+  } = editingOptions;
+
+  const onCommitChanges = React.useCallback(({ added, changed, deleted }) => {
+    if (added) {
+      const startingAddedId = data.length > 0 ? data[data.length - 1].id + 1 : 0;
+      setData([...data, { id: startingAddedId, ...added }]);
+    }
+    if (changed) {
+      setData(data.map(appointment => (
+        changed[appointment.id] ? { ...appointment, ...changed[appointment.id] } : appointment)));
+    }
+    if (deleted !== undefined) {
+      setData(data.filter(appointment => appointment.id !== deleted));
+    }
+    setIsAppointmentBeingCreated(false);
+  }, [setData, setIsAppointmentBeingCreated]);
+  const onAddedAppointmentChange = React.useCallback((appointment) => {
+    setAddedAppointment(appointment);
+    setIsAppointmentBeingCreated(true);
+  });
+  const handleEditingOptionsChange = React.useCallback(({ target }) => {
+    const { value } = target;
+    const { [value]: checked } = editingOptions;
+    setEditingOptions({
+      ...editingOptions,
+      [value]: !checked,
+    });
+  });
+
+  const TimeTableCell = React.useCallback(React.memo(({ onDoubleClick, ...restProps }) => (
+    <WeekView.TimeTableCell
+      {...restProps}
+      onDoubleClick={allowAdding ? onDoubleClick : undefined}
+    />
+  )), [allowAdding]);
+
+  const CommandButton = React.useCallback(({ id, ...restProps }) => {
+    if (id === 'deleteButton') {
+      return <AppointmentForm.CommandButton id={id} {...restProps} disabled={!allowDeleting} />;
+    }
+    return <AppointmentForm.CommandButton id={id} {...restProps} />;
+  }, [allowDeleting]);
+
+  const allowDrag = React.useCallback(
+    () => allowDragging && allowUpdating, [allowDragging, allowUpdating],
+  );
+  const allowResize = React.useCallback(
+    () => allowResizing && allowUpdating, [allowResizing, allowUpdating],
+  );
+
+  return (
+    <>
+      <EditingOptionsSelector
+        editingOptions={editingOptions}
+        handleEditingOptionsChange={handleEditingOptionsChange}
+      />
```

Comment:
The plugin name is `EditingOptionsSelector`. The properties should be `options` and `onOptionsChanged` without the `editing` world.